src/coreComponents/fileIO/timeHistory/HDFFile.cpp

```diff
@@ -82,12 +82,12 @@ HDFFile::~HDFFile()
   H5Fclose( m_fileId );
 }
 
-bool HDFFile::checkInTarget( const string & name ) const
+bool HDFFile::hasDataset( const string & name ) const
 {
   int exists = 0;
   H5E_BEGIN_TRY
   {
-    exists = H5Gget_objinfo( this->operator int64_t(), name.c_str(), 0, NULL );
+    exists = H5Gget_objinfo( *this, name.c_str(), 0, NULL );
```

Comment:
is `this` implicitly convertible to `int64_t` or some `hid_t` typedef or is the called function changed ?

Reply:
`this` is convertible to `int64_t ` thanks to its `operator int64_t()`.